### ISSUE TITLE
ImpureMethodCall false-positive for Collection::toArray()

### ISSUE THREAD

**Issue by fluffycondor** [weirdan/doctrine-psalm-plugin#85]
```php
class Test {
    
    private Collection $collection;
    
    /**
     * @psalm-mutation-free
     */
    public function test(): array
    {
        return $this->collection->toArray();
    }
}
```

ImpureMethodCall: Cannot call an possibly-mutating method Doctrine\Common\Collections\Collection::toArray from a mutation-free context

doctrine/orm 2.7.4
weirdan/doctrine-psalm-plugin 0.11.3
vimeo/psalm 3.18.2

**Comment by weirdan** [comment]
Is it implied to be mutation-free? I'd suggest to ask `doctrine/collection` devs (they are not very responsive though).
